File: src/main/java/org/example/kafka/HelloController.java
package org.example.kafka;

import io.micronaut.http.MediaType;
import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Get;

import javax.inject.Inject;

@Controller("/hello")
public class HelloController {

    @Inject
    TestProducer producer;

    @Get(produces = MediaType.TEXT_PLAIN)
    public String index() {

        producer.send("aKey", "Hello world!");

        return "Hello World";
    }
}

File: src/main/kotlin/org/example/t/HelloController.kt
package org.example.t

import io.micronaut.http.MediaType
import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Get
import org.example.kafka.TestProducer
import javax.inject.Inject

@Controller("/hello")
class HelloController {

    @Inject
    internal var producer: TestProducer? = null

    @Get(produces = arrayOf(MediaType.TEXT_PLAIN))
    fun index(): String {

        producer!!.send("aKey", "Hello Kotlin world!")

        return "Hello Kotlin World!!"
    }
}